sqlglot/expressions.py

```diff
@@ -1006,6 +1006,31 @@ class Create(Expression):
         "clone": False,
     }
 
+    @property
+    def ctes(self):
+        with_ = self.args.get("with")
+        if not with_:
+            return []
+        return with_.expressions
+
+    @property
+    def named_selects(self) -> t.List[str]:
+        if isinstance(self.expression, Select):
+            return self.expression.named_selects
+        return []
+
+    @property
+    def is_star(self) -> bool:
+        if isinstance(self.expression, Select):
+            return self.expression.is_star
+        return False
+
+    @property
+    def selects(self) -> t.List[Expression]:
+        if isinstance(self.expression, Select):
```

Comment:
this seems too specific, should it instead be like checking subqueryable, derived_table or has the attribute selects?

Reply:
all the isinstance checks i think should be more general